backend/users/role_access.py

```
@@ -0,0 +1,53 @@
+
+
+def user_has_role_profile(user, category):
+    """Return whether the user holds the real profile for a portal role."""
+    user_id = getattr(user, 'pk', None)
+    if not user_id or category not in VIEWABLE_ROLE_CATEGORIES:
+        return False
+
+    cache = getattr(user, '_role_profile_access_cache', None)
+    if cache is None:
+        cache = {}
+        setattr(user, '_role_profile_access_cache', cache)
```

Ruff (0.15.21)

[warning] Do not call `setattr` with a constant attribute value. It is not any safer than normal property access.

Replace `setattr` with assignment

(B010)